components/search/CustomSearchBox.tsx

```diff
@@ -44,4 +48,14 @@ function SearchBox({ refine }: { refine: Dispatch<SetStateAction<string>> }) {
   );
 }
 
+function connectSearchBox(Component: any) {
+  const SearchBox = (props: any) => {
+    const data = useSearchBox(props);
```

Comment:
Similar story here - `useSearchBox` replaced `connectSearchBox`. This is lightly adapted from their [documentation](https://www.algolia.com/doc/guides/building-search-ui/upgrade-guides/react/#connectors).